tests/test_resources.py

```diff
@@ -213,68 +324,143 @@ def test_user_albums(self, client):
+        artist = next(artists_generator)
+        assert isinstance(artist, deezer.resources.Artist)
+        assert repr(artist) == "<Artist: Wax Tailor>"
 
     def test_user_playlists(self, client):
         """
         Test playlists method of user resource
         """
         user = client.get_user(359622)
+
+        # tests list
         playlists = user.get_playlists()
         assert isinstance(playlists, list)
         playlist = playlists[0]
         assert isinstance(playlist, deezer.resources.Playlist)
         assert repr(playlist) == "<Playlist: AC/DC>"
-        assert type(user.iter_playlists()) == GeneratorType
+
+        # tests generator
+        playlists_generator = user.iter_playlists()
+        assert type(playlists_generator) == GeneratorType
+        playlist = next(playlists_generator)
+        assert isinstance(playlist, deezer.resources.Playlist)
+        assert repr(playlist) == "<Playlist: AC/DC>"
 
     def test_user_tracks(self, client):
         """
         Test tracks method of user resource
         """
         user = client.get_user(353978015)
+
+        # tests list
         tracks = user.get_tracks()
         assert isinstance(tracks, list)
         track = tracks[0]
         assert isinstance(track, deezer.resources.Track)
         assert repr(track) == "<Track: Prélude a l'après-midi d'un faune, L. 86>"
-        assert type(user.iter_tracks()) == GeneratorType
+
+        # tests generator
+        tracks_generator = user.iter_tracks()
+        assert type(tracks_generator) == GeneratorType
+        track = next(tracks_generator)
+        assert isinstance(track, deezer.resources.Track)
+        assert repr(track) == "<Track: Prélude a l'après-midi d'un faune, L. 86>"
 
 
 class TestPlaylist:
     def test_get_tracks(self, client):
+        """
+        Test tracks method of playlist resource
+        """
         playlist = client.get_playlist(12345)
+
+        # tests list
         tracks = playlist.get_tracks()
+        assert isinstance(tracks, list)
         assert len(tracks) == 4
         for track in tracks:
             assert isinstance(track, deezer.resources.Track)
         assert tracks[0].title == "Skanky Panky"
-        assert type(playlist.iter_tracks()) == GeneratorType
+
+        # tests generator
+        tracks_generator = playlist.iter_tracks()
+        assert type(tracks_generator) == GeneratorType
+        track = next(tracks_generator)
+        assert track.title == "Skanky Panky"
+        count = 1
+        while 1:
+            assert isinstance(track, deezer.resources.Track)
+            try:
+                track = next(tracks_generator)
+                count += 1
+            except StopIteration:
+                break
+        # this is weird (the count being different), it seems there's a sort of
+        # partially-hidden track that only shows up when using the method in
+        # iter_tracks
+        assert count == 5
```

Comment:
That's weird, I looked at the content of the response and it was:

```
[<Track: Skanky Panky>,
 <Track: More Dance Music>,
 <Track: Basin Street Blues>,
 <Track: Celebrity Death Match (feat. Kid Koala)>,
 <Track: Celebrity Death Match (feat. Kid Koala)>]
```

There is potentially a bug somewhere, but it's strange that it doesn't affect others `iter_...` methods, as the Playlist's implementation hasn't got much specific.

Reply:
This is a bug with the Deezer API itself (I think). The final result is gained by `iter_tracks` performing a second call to the endpoint, this time with `index=4`. I've no idea why it happens, given that the correct index (for the number of previous results) is being passed to the API.

I'm afraid I probably won't be much use in solving this one!

Reply:
Yea, no need to worry too much about this one, looks out of scope anyway.